From c16942e81415d7ae06640250bf78e639390eb5e0 Mon Sep 17 00:00:00 2001
From: Tony Cheng <tony.cheng@amd.com>
Date: Thu, 25 Aug 2016 20:54:57 -0400
Subject: [PATCH 0937/1722] drm/amd/dal: remove obsolete bios parser code

- remove detect_sink.

  Detect sink is used in vga load detection which DAL3 don't support yet.
  Only ASIC in DAL3 that support VGA is Bonaire so if load detect is necessary
  it might be easier to achieve by just programming the necessary DAC registers
  ourselves.

- remove dac_load_detection

Change-Id: I9db6c98d4c420a4f5fd0cae7e6f956a2689fbd1f
Signed-off-by: Jordan Lazare <Jordan.Lazare@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c      | 105 +++++++++++++++++++-
 .../gpu/drm/amd/dal/dc/bios/bios_parser_helper.h   |   7 --
 drivers/gpu/drm/amd/dal/dc/bios/command_table.c    | 109 ---------------------
 drivers/gpu/drm/amd/dal/dc/bios/command_table.h    |   5 -
 .../dal/dc/bios/dce110/bios_parser_helper_dce110.c |   1 -
 .../dal/dc/bios/dce112/bios_parser_helper_dce112.c |   1 -
 .../dal/dc/bios/dce80/bios_parser_helper_dce80.c   |   1 -
 .../gpu/drm/amd/dal/include/bios_parser_types.h    |   1 -
 8 files changed, 103 insertions(+), 127 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
index d4ba3f0..6e76aaa 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser.c
@@ -4330,9 +4330,110 @@ void dc_bios_destroy_integrated_info(struct dc_bios *dcb,
 	}
 }
 
-/******************************************************************************
+/******************************************************************************/
+
+static const struct dc_vbios_funcs vbios_funcs = {
+        .get_connectors_number = bios_parser_get_connectors_number,
+
+        .power_up = bios_parser_power_up,
+
+        .get_encoders_number = bios_parser_get_encoders_number,
+
+        .get_oem_ddc_lines_number = bios_parser_get_oem_ddc_lines_number,
+
+        .get_encoder_id = bios_parser_get_encoder_id,
+
+        .get_connector_id = bios_parser_get_connector_id,
+
+        .get_dst_number = bios_parser_get_dst_number,
+
+        .get_gpio_record = bios_parser_get_gpio_record,
+
+        .get_src_obj = bios_parser_get_src_obj,
+
+        .get_dst_obj = bios_parser_get_dst_obj,
+
+        .get_i2c_info = bios_parser_get_i2c_info,
+
+        .get_oem_ddc_info = bios_parser_get_oem_ddc_info,
+
+        .get_voltage_ddc_info = bios_parser_get_voltage_ddc_info,
+
+        .get_thermal_ddc_info = bios_parser_get_thermal_ddc_info,
+
+        .get_hpd_info = bios_parser_get_hpd_info,
+
+        .get_device_tag = bios_parser_get_device_tag,
+
+        .get_firmware_info = bios_parser_get_firmware_info,
+
+        .get_spread_spectrum_info = bios_parser_get_spread_spectrum_info,
+
+        .get_ss_entry_number = bios_parser_get_ss_entry_number,
+
+        .get_embedded_panel_info = bios_parser_get_embedded_panel_info,
+
+        .get_gpio_pin_info = bios_parser_get_gpio_pin_info,
+
+        .get_embedded_panel_info = bios_parser_get_embedded_panel_info,
+
+        .get_gpio_pin_info = bios_parser_get_gpio_pin_info,
+
+        .get_encoder_cap_info = bios_parser_get_encoder_cap_info,
+
+        .get_din_connector_info = bios_parser_get_din_connector_info,
+
+        .is_accelerated_mode = bios_parser_is_accelerated_mode,
+
+        .set_scratch_critical_state = bios_parser_set_scratch_critical_state,
+
+        .is_device_id_supported = bios_parser_is_device_id_supported,
+	
+	 /* COMMANDS */
+        .encoder_control = bios_parser_encoder_control,
+
+        .transmitter_control = bios_parser_transmitter_control,
+
+        .crt_control = bios_parser_crt_control,
+
+        .enable_crtc = bios_parser_enable_crtc,
+
+        .adjust_pixel_clock = bios_parser_adjust_pixel_clock,
+
+        .set_pixel_clock = bios_parser_set_pixel_clock,
+
+        .set_dce_clock = bios_parser_set_dce_clock,
+
+        .enable_spread_spectrum_on_ppll = bios_parser_enable_spread_spectrum_on_ppll,
+
+        .program_crtc_timing = bios_parser_program_crtc_timing,
+
+        .blank_crtc = bios_parser_blank_crtc,
+
+        .set_overscan = bios_parser_set_overscan,
+
+        .crtc_source_select = bios_parser_crtc_source_select,
+
+        .program_display_engine_pll = bios_parser_program_display_engine_pll,
+
+        .get_divider_for_target_display_clock = bios_parser_get_divider_for_target_display_clock,
+
+        .enable_memory_requests = bios_parser_enable_memory_requests,
+
+        .external_encoder_control = bios_parser_external_encoder_control,
+
+        .enable_disp_power_gating = bios_parser_enable_disp_power_gating,
+
+        .post_init = bios_parser_post_init,
+
+        .create_integrated_info = bios_parser_create_integrated_info,
+
+        .destroy_integrated_info = bios_parser_destroy_integrated_info,
+
+        .bios_parser_destroy = bios_parser_destroy,
+};
 
- * Stub-functions */
+/* Stub-functions */
 
 static bool bios_parser_construct(
 	struct bios_parser *bp,
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.h b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.h
index 2f8c540..1815acca 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.h
+++ b/drivers/gpu/drm/amd/dal/dc/bios/bios_parser_helper.h
@@ -41,13 +41,6 @@
 struct bios_parser;
 
 struct bios_parser_helper {
-	enum signal_type (*detect_sink)(
-		struct dc_context *ctx,
-		struct graphics_object_id encoder,
-		struct graphics_object_id connector,
-		enum signal_type signal);
-	void (*set_scratch_acc_mode_change)(
-		struct dc_context *ctx);
 	bool (*is_accelerated_mode)(
 		struct dc_context *ctx);
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
index 81fbb47..8701208 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table.c
@@ -54,7 +54,6 @@ static void init_enable_spread_spectrum_on_ppll(struct bios_parser *bp);
 static void init_adjust_display_pll(struct bios_parser *bp);
 static void init_dac_encoder_control(struct bios_parser *bp);
 static void init_dac_output_control(struct bios_parser *bp);
-static void init_dac_load_detection(struct bios_parser *bp);
 static void init_blank_crtc(struct bios_parser *bp);
 static void init_set_crtc_timing(struct bios_parser *bp);
 static void init_set_crtc_overscan(struct bios_parser *bp);
@@ -76,7 +75,6 @@ void dal_bios_parser_init_cmd_tbl(struct bios_parser *bp)
 	init_adjust_display_pll(bp);
 	init_dac_encoder_control(bp);
 	init_dac_output_control(bp);
-	init_dac_load_detection(bp);
 	init_blank_crtc(bp);
 	init_set_crtc_timing(bp);
 	init_set_crtc_overscan(bp);
@@ -1690,94 +1688,6 @@ static enum bp_result dac2_output_control_v1(
 /*******************************************************************************
  ********************************************************************************
  **
- **                  DAC LOAD DETECTION
- **
- ********************************************************************************
- *******************************************************************************/
-
-static enum signal_type dac_load_detection_v3(
-	struct bios_parser *bp,
-	struct graphics_object_id encoder,
-	struct graphics_object_id connector,
-	enum signal_type display_signal);
-
-static void init_dac_load_detection(struct bios_parser *bp)
-{
-	switch (BIOS_CMD_TABLE_PARA_REVISION(DAC_LoadDetection)) {
-	case 3:
-		bp->cmd_tbl.dac_load_detection = dac_load_detection_v3;
-		break;
-	default:
-		bp->cmd_tbl.dac_load_detection = NULL;
-		break;
-	}
-}
-
-static enum signal_type dac_load_detection_v3(
-	struct bios_parser *bp,
-	struct graphics_object_id encoder,
-	struct graphics_object_id connector,
-	enum signal_type display_signal)
-{
-	DAC_LOAD_DETECTION_PS_ALLOCATION params;
-	enum signal_type signal = SIGNAL_TYPE_NONE;
-
-	memset(&params, 0, sizeof(params));
-
-	/* load detection is cupported for CRT, TV and CV */
-	switch (display_signal) {
-	case SIGNAL_TYPE_RGB:
-		switch (dal_graphics_object_id_get_encoder_id(encoder)) {
-		case ENCODER_ID_INTERNAL_DAC1:
-		case ENCODER_ID_INTERNAL_KLDSCP_DAC1:
-			params.sDacload.usDeviceID =
-				cpu_to_le16(ATOM_DEVICE_CRT1_SUPPORT);
-			break;
-		case ENCODER_ID_INTERNAL_DAC2:
-		case ENCODER_ID_INTERNAL_KLDSCP_DAC2:
-			params.sDacload.usDeviceID =
-				cpu_to_le16(ATOM_DEVICE_CRT2_SUPPORT);
-			break;
-		default:
-			break;
-		}
-		break;
-		default:
-			return signal;
-	}
-
-	/* set the encoder to detect on */
-	switch (dal_graphics_object_id_get_encoder_id(encoder)) {
-	case ENCODER_ID_INTERNAL_DAC1:
-	case ENCODER_ID_INTERNAL_KLDSCP_DAC1:
-		params.sDacload.ucDacType = ATOM_DAC_A;
-		break;
-	case ENCODER_ID_INTERNAL_DAC2:
-	case ENCODER_ID_INTERNAL_KLDSCP_DAC2:
-		params.sDacload.ucDacType = ATOM_DAC_B;
-		break;
-	default:
-		return signal;
-	}
-
-	if (!EXEC_BIOS_CMD_TABLE(DAC_LoadDetection, params))
-		return signal;
-#if defined(CONFIG_DRM_AMD_DAL_VBIOS_PRESENT)
-	signal = bp->bios_helper->detect_sink(
-			bp->ctx,
-			encoder,
-			connector,
-			display_signal);
-#else
-	BREAK_TO_DEBUGGER(); /* VBios is needed */
-#endif
-
-	return signal;
-}
-
-/*******************************************************************************
- ********************************************************************************
- **
  **                 BLANK CRTC
  **
  ********************************************************************************
@@ -2568,25 +2478,6 @@ static enum bp_result external_encoder_control_v3(
 	if (EXEC_BIOS_CMD_TABLE(ExternalEncoderControl, params))
 		result = BP_RESULT_OK;
 
-	if (EXTERNAL_ENCODER_CONTROL_DAC_LOAD_DETECT == cntl->action) {
-#if defined(CONFIG_DRM_AMD_DAL_VBIOS_PRESENT)
-		if (BP_RESULT_OK == result)
-			/* get VBIOS result from scratch register.
-			 * ExternalEncoderControl runs detection and save result
-			 * in BIOS scratch registers. */
-			cntl->signal = bp->bios_helper->detect_sink(
-					bp->ctx,
-					encoder,
-					cntl->connector_obj_id,
-					cntl->signal);
-		else/* BIOS table does not work. */
-#endif
-		{
-			BREAK_TO_DEBUGGER(); /* VBios is needed */
-			cntl->signal = SIGNAL_TYPE_NONE;
-		}
-	}
-
 	return result;
 }
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/command_table.h b/drivers/gpu/drm/amd/dal/dc/bios/command_table.h
index 3cb0c7f..e1cd21b 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/command_table.h
+++ b/drivers/gpu/drm/amd/dal/dc/bios/command_table.h
@@ -68,11 +68,6 @@ struct cmd_tbl {
 	enum bp_result (*dac2_output_control)(
 		struct bios_parser *bp,
 		bool enable);
-	enum signal_type (*dac_load_detection)(
-		struct bios_parser *bp,
-		struct graphics_object_id encoder,
-		struct graphics_object_id connector,
-		enum signal_type display_signal);
 	enum bp_result (*blank_crtc)(
 		struct bios_parser *bp,
 		struct bp_blank_crtc_parameters *bp_params,
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c b/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
index d398291..7147fe7 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/dce110/bios_parser_helper_dce110.c
@@ -170,7 +170,6 @@ void dce110_set_scratch_critical_state(struct dc_context *ctx,
 
 /* function table */
 static const struct bios_parser_helper bios_parser_helper_funcs = {
-	.detect_sink = detect_sink,
 	.is_accelerated_mode = is_accelerated_mode,
 };
 
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/dce112/bios_parser_helper_dce112.c b/drivers/gpu/drm/amd/dal/dc/bios/dce112/bios_parser_helper_dce112.c
index a00c905..84296da 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/dce112/bios_parser_helper_dce112.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/dce112/bios_parser_helper_dce112.c
@@ -154,7 +154,6 @@ static enum signal_type detect_sink(
 
 /* function table */
 static const struct bios_parser_helper bios_parser_helper_funcs = {
-	.detect_sink = detect_sink,
 	.is_accelerated_mode = is_accelerated_mode,
 	.set_scratch_acc_mode_change = set_scratch_acc_mode_change
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/bios/dce80/bios_parser_helper_dce80.c b/drivers/gpu/drm/amd/dal/dc/bios/dce80/bios_parser_helper_dce80.c
index 8f55d67..fcddd79 100644
--- a/drivers/gpu/drm/amd/dal/dc/bios/dce80/bios_parser_helper_dce80.c
+++ b/drivers/gpu/drm/amd/dal/dc/bios/dce80/bios_parser_helper_dce80.c
@@ -130,7 +130,6 @@ static bool is_accelerated_mode(
 }
 
 static const struct bios_parser_helper bios_parser_helper_funcs = {
-	.detect_sink = detect_sink,
 	.is_accelerated_mode = is_accelerated_mode,
 };
 
diff --git a/drivers/gpu/drm/amd/dal/include/bios_parser_types.h b/drivers/gpu/drm/amd/dal/include/bios_parser_types.h
index f986012..726cd23 100644
--- a/drivers/gpu/drm/amd/dal/include/bios_parser_types.h
+++ b/drivers/gpu/drm/amd/dal/include/bios_parser_types.h
@@ -82,7 +82,6 @@ enum bp_external_encoder_control_action {
 	EXTERNAL_ENCODER_CONTROL_SETUP = 0xf,
 	EXTERNAL_ENCODER_CONTROL_UNBLANK = 0x10,
 	EXTERNAL_ENCODER_CONTROL_BLANK = 0x11,
-	EXTERNAL_ENCODER_CONTROL_DAC_LOAD_DETECT = 0x12
 };
 
 enum bp_pipe_control_action {
-- 
2.7.4

